unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx

```diff
@@ -40,19 +54,169 @@ interface DisconnectStripeProps {
+    lockAddress,
+    network,
+  })
+
+  const {
+    mutateAsync: saveSettingMutation,
+    isLoading: isSaveLockSettingLoading,
+  } = useSaveLockSettings()
+
+  const { data: { data: lockSettings = {} } = {}, isLoading } =
+    useGetLockSettings({
+      lockAddress,
+      network,
+    })
+
+  useEffect(() => {
+    const price = lockSettings?.creditCardPrice
+    if (!price) return
+    setValue('creditCardPrice', price)
+    setUseCustomPrice(true)
+  }, [lockSettings?.creditCardPrice, setValue])
+
+  const hasPriceConversion = fiatPricing?.usd?.keyPrice
+
+  const onSaveCreditCardPrice = async ({
+    creditCardPrice,
+  }: CreditCardFormSchema) => {
+    const price = creditCardPrice ? parseFloat(`${creditCardPrice}`) : null
+
+    const savePricePromise = saveSettingMutation({
+      lockAddress,
+      network,
+      creditCardPrice: price,
+    })
+
+    await ToastHelper.promise(savePricePromise, {
+      loading: 'Updating price...',
+      success: 'Price updated.',
+      error: 'There is some issue updating the price.',
+    })
+  }
+
+  const onSubmit = (fields: CreditCardFormSchema) => {
+    onSaveCreditCardPrice(fields)
+  }
+
+  if (isLoading) {
+    return (
+      <Placeholder.Root>
+        <Placeholder.Card />
+      </Placeholder.Root>
+    )
+  }
+
+  const saveDisabled = isSaveLockSettingLoading || disabled
+
+  const needsCustomPrice = !hasPriceConversion && !lockSettings?.creditCardPrice
+  const symbol = lock?.currencySymbol
+
+  return (
+    <div className="grid gap-1">
+      <SettingCardDetail title="Price in $USD" />
+      <div className="grid grid-cols-1 gap-1 p-4 bg-gray-100 rounded-lg">
+        <form
+          onSubmit={handleSubmit(onSubmit)}
+          className="grid items-center gap-3"
+        >
+          {hasPriceConversion && (
+            <>
+              <ToggleSwitch
+                title="Use fixed price"
+                enabled={useCustomPrice}
+                setEnabled={setUseCustomPrice}
+                onChange={(enabled) => {
+                  if (!enabled) {
+                    setValue('creditCardPrice', null, {
+                      shouldValidate: true,
+                      shouldDirty: true,
+                    })
+                  }
+                }}
+              />
+              {!useCustomPrice && (
+                <span className="text-sm">
+                  When custom price is not set, the conversion price will be
+                  used for credit card payments.
+                </span>
+              )}
+            </>
+          )}
+          <>
+            {(useCustomPrice || !hasPriceConversion) && (
+              <Input
+                type="numeric"
+                step="any"
+                disabled={disabled}
+                description={
+                  hasPriceConversion
+                    ? 'Price in $USD that will be charged, the actual conversion price will ignored.'
+                    : `Price in $USD that will be charged`
```

Comment:
```suggestion
                  hasPriceConversion
                    ? 'Price in $USD that will be charged, the actual conversion price will ignored.'
                    : `Price in $USD that will be charged`
```

@julien51 can we improve copy here?